### PR TITLE
Add more test for login links and fix a bug on action retrieve

### PR DESCRIPTION
**Pull request by lunny**
As title.

### CODE REVIEW

File: models/action.go [hunk 1]
```diff
@@ -735,7 +735,7 @@ func GetFeeds(opts GetFeedsOptions) ([]*Action, error) {
 
 	if opts.Collaborate {
 		cond = builder.Eq{"user_id": opts.RequestedUser.ID}.Or(
-			builder.Expr(`repo_id IN (SELECT repo_id FROM "access" WHERE access.user_id = ?)`, opts.RequestedUser.ID))
+			builder.Expr("repo_id IN (SELECT repo_id FROM `access` WHERE access.user_id = ?)", opts.RequestedUser.ID))
```
Comment (lafriks) [inline models/action.go]: Why such change in this PR?

Comment (lunny) [reply models/action.go]: I find this bug via these tests.

**Comment by lafriks** [comment]
There are conflicts now

**Comment by lunny** [comment]
@lafriks rebased now.

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Comment by lafriks** [comment]
Integration test failed

**Comment by lafriks** [comment]
LGTM

File: integrations/links_test.go [hunk 2]
```diff
@@ -24,10 +27,104 @@ func TestLinksNoLogin(t *testing.T) {
...
+	}
+
+	session := loginUser(t, userName)
+	for _, link := range links {
+		req := NewRequest(t, "GET", link)
+		session.MakeRequest(t, req, http.StatusOK)
+	}
+
+	reqAPI := NewRequestf(t, "GET", "/api/v1/users/%s/repos", userName)
+	respAPI := MakeRequest(t, reqAPI, http.StatusOK)
```
Comment (Morlinest) [inline integrations/links_test.go:91]: Shouldn't be here `session.MakeRequest`?

File: integrations/links_test.go [hunk 3]
```diff
@@ -24,10 +27,104 @@ func TestLinksNoLogin(t *testing.T) {
...
+		"/settings/keys",
+		"/releases",
+		"/releases/new",
+		//"/wiki/_pages",
+		"/wiki/_new",
+	}
+
+	for _, repo := range apiRepos {
+		for _, link := range repoLinks {
+			req := NewRequest(t, "GET", fmt.Sprintf("/%s/%s%s", userName, repo.Name, link))
```
Comment (Morlinest) [inline integrations/links_test.go]: `NewRequestf` instead of `fmt.Sprintf`?

Comment (Morlinest) [inline integrations/links_test.go:120]: `NewRequestf` instead of `fmt.Sprintf`?

**Pull request merged**